lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/security/LocalFileBasedAccessControlRulesProvider.java

```diff
@@ -0,0 +1,26 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.security;
+
+import com.google.inject.Inject;
+
+public class LocalFileBasedAccessControlRulesProvider
+        extends LocalFileAccessControlRulesProvider<AccessControlRules>
```

Comment:
Why do you need this inheritance?

Reply:
I use it in `FileBasedAccessControlModule` for binding it. Do you see a better way of binding it without subclassing which takes into account that the generic type must be bound in the constructor via the second parameter ?

Reply:
```
            binder.bind(new TypeLiteral<Supplier<AccessControlRules>>() {})
                    .toProvider(() -> new LocalFileAccessControlRulesProvider(configuration, AccessControlRules.class));
```
?